tests/pytorch/fused_attn/test_fused_attn.py

```diff
@@ -177,11 +177,11 @@ def _is_flash_attention_supported(config: ModelConfig) -> bool:
         return False
     if config.num_heads != config.num_gqa_groups and not _is_flash_attention_2_available():
         return False
-    if "causal" in config.attn_mask_type and config.attn_type == "cross":
-        if _is_flash_attention_2_1():
-            # FAv2.1 implements causal mask for cross attention differently
-            # https://github.com/Dao-AILab/flash-attention#21-change-behavior-of-causal-flag
-            return False
+    # if "causal_bottom_right" in config.attn_mask_type and config.attn_type == "cross":
+    #    if _is_flash_attention_2_1():
+    #        # FAv2.1 implements causal mask for cross attention differently
+    #        # https://github.com/Dao-AILab/flash-attention#21-change-behavior-of-causal-flag
+    #        return False
```

Comment:
Debugging code?
```suggestion
    if "causal" in config.attn_mask_type and config.attn_type == "cross":
       if _is_flash_attention_2_1():
           # FAv2.1 implements causal mask for cross attention differently
           # https://github.com/Dao-AILab/flash-attention#21-change-behavior-of-causal-flag
           return False
```
Also, do we need a similar check to disable bottom-right causal for FA <2.1?

Reply:
Ya, a lot of debugging code in that commit.

Code is refactored, and a check for FA 2.1 is add [here](https://github.com/cyanguwa/TransformerEngine/blob/4d2c55a1ae087e5c7409c08701d66a2eaff9af0f/transformer_engine/pytorch/attention.py#L352).